aria/modeling/service_instance.py

```diff
@@ -180,6 +181,16 @@ def validate_capabilities(self):
                 satisfied = False
         return satisfied
 
+    def find_hosts(self):
```

Comment:
That's not true: this phase will need be run every time there is a change in the topology. This is part of the reason I don't think `instantiation` is the right name for this module we talk about, since it would actually be used in many cases after instantiation.